### PR TITLE
Update Mac environment files to be more human-readable

### PR DESCRIPTION
**Pull request by garrettmflynn**
This may be more a reflection of my own experience than a general experience for developers of the GUIDE, but I'm continually choosing the wrong Mac build for my system (`MAC-arm64`) when resetting the Python environment.

To guard against this in the future, I've changed the Mac naming convention to `MAC-intel` vs. `MAC-apple` for a more human-readable distinction between the two systems.

### CODE REVIEW

**Comment by rly** [comment]
`MAC-apple` seems like it might also be confusing to the average user. How about `MAC-apple-silicon`?

Should we also change the distributable names away from `NWB-GUIDE-x64.dmg` and `NWB-GUIDE-arm64.dmg`?

**Comment by CodyCBakerPhD** [comment]
> Should we also change the distributable names away from NWB-GUIDE-x64.dmg and NWB-GUIDE-arm64.dmg?

I don't know how to do that - do you? We just put the architecture identifier in there from the electron builder

File: .github/workflows/pyflask-build-and-dist-tests.yml [hunk 1]
```diff
@@ -33,12 +33,12 @@ jobs:
 
           - python-version: "3.9"
             os: macos-latest  # Mac arm64 runner
-            label: environments/environment-MAC-arm64.yml
+            label: environments/environment-MAC-apple.yml
```
Comment (CodyCBakerPhD) [inline .github/workflows/pyflask-build-and-dist-tests.yml:36]:
```suggestion
            label: environments/environment-MAC-apple-silicon.yml
```

File: .github/workflows/testing-live-services.yml [hunk 2]
```diff
@@ -28,10 +28,10 @@ jobs:
             label: environments/environment-Linux.yml
 
           - os: macos-latest  # Mac arm64 runner
-            label: environments/environment-MAC-arm64.yml
+            label: environments/environment-MAC-apple.yml
```
Comment (CodyCBakerPhD) [inline .github/workflows/testing-live-services.yml:31]:
```suggestion
            label: environments/environment-MAC-silicon.yml
```

File: .github/workflows/testing.yml [hunk 3]
```diff
@@ -28,10 +28,10 @@ jobs:
             label: environments/environment-Linux.yml
 
           - os: macos-latest  # Mac arm64 runner
-            label: environments/environment-MAC-arm64.yml
+            label: environments/environment-MAC-apple.yml
```
Comment (CodyCBakerPhD) [inline .github/workflows/testing.yml:31]:
```suggestion
            label: environments/environment-MAC-silicon.yml
```

File: docs/developer_guide.rst [hunk 4]
```diff
@@ -36,13 +36,13 @@ Install the appropriate Python dependencies for your operating system.
...
 
-    conda env create -f ./environments/environment-MAC.yml
+    conda env create -f ./environments/environment-MAC-intel.yml
 
 **Mac with arm64 architecture**
 
 .. code-block:: bash
 
-    conda env create -f ./environments/environment-MAC-arm64.yml
+    conda env create -f ./environments/environment-MAC-apple.yml
```
Comment (CodyCBakerPhD) [inline docs/developer_guide.rst:45]:
```suggestion
    conda env create -f ./environments/environment-MAC-apple-silicon.yml
```

**Comment by garrettmflynn** [comment]
Updated programmatically for consistency. 

> I don't know how to do that - do you? We just put the architecture identifier in there from the electron builder

AFAIK this is handled by `electron-builder` itself based on the architectures you've specified to build (e.g. `arm64`, `x64`, etc.) so we can't control the output filename of the `.dmg`

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**